lxd/db/cluster/profiles.go

```diff
@@ -41,3 +48,42 @@ type ProfileFilter struct {
 	Project *string
 	Name    *string
 }
+
+// ToAPI returns a cluster Profile as an API struct.
+func (p *Profile) ToAPI(ctx context.Context, tx *sql.Tx) (*api.Profile, error) {
+	config, err := GetProfileConfig(ctx, tx, p.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	devices, err := GetProfileDevices(ctx, tx, p.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	profile := &api.Profile{
+		Name: p.Name,
+		ProfilePut: api.ProfilePut{
+			Description: p.Description,
+			Config:      config,
+			Devices:     DevicesToAPI(devices),
```

Comment:
Is there a reason we have duplicated `DevicesToAPI` function in cluster and db packages?

Reply:
`Instance` structs aren't using the updated generator logic, so those generated methods are going to look for `db.Device` instead of `cluster.Device`. I think this is the last vestige of `db.Device`, so once `Instance` is transitioned (which is coming up next) this duplication should be gone.